ci: include more tests in windows ci

alan-agius4 opened the pull request:
**test: fix paths tests to work cross platform**
 In Windows when `/test.txt` is resolved it will be resolved to `[DRIVE]:/test.txt`
 
**ci: update windows docker image for codefresh**
 This commit changes a couple of things

1) Sets the escape character to backtick. Setting the escape character to backtick is especially useful on Windows where `\` is the directory path separator.
2) Install Python since it is a requirement for certain bazel tool chains that are now used.
3) Install VS Build Tools with C++ components as C++ compilation is required for certain bazel tools chains.
4) Removes Visual C++ Redistributable as it seems that is no longer required because of the VS Builds Tools
 
**ci: include more tests in windows ci**
 We added most of the tests excluded the once which are tagged with `browser:chromium-local` as they are currently not working in CI but work locally.

The `--test_tag_filters` option is now passed via command line as it seems that it has no effect it is passed in both RC and command line. IE command line options are not merged with the RC options but rather overridden.

alan-agius4 committed:
test: fix paths tests to work cross platform

In Windows when `/test.txt` is resolved it will be resolved to `[DRIVE]:/test.txt`

alan-agius4 committed:
ci: update windows docker image for codefresh

This commit changes a couple of things

1) Sets the escape character to backtick. Setting the escape character to backtick is especially useful on Windows where `\` is the directory path separator.
2) Install Python since it is a requirement for certain bazel tool chains that are now used.
3) Install VS Build Tools with C++ components as they C++ compilation is required for certain bazel tools chains.
4) Removes Visual C++ Redistributable as it seems that is no longer required because of the VS Builds Tools

alan-agius4 commented on .codefresh/codefresh.yml:
```diff
@@ -11,14 +11,29 @@ steps:
+  Tests:
+    type: parallel
+    steps:
+      RunVETests:
```
These parallel tasks are not optimised. But CI seems to perform better this way

gkalpak commented on .codefresh/codefresh.yml:
```diff
@@ -12,13 +12,16 @@ steps:
     # Run tests
-    - yarn test-ivy-aot //packages/animations/test //packages/common/test //packages/forms/test //packages/http/test //packages/platform-browser/test //packages/platform-browser-dynamic/test //packages/router/test
-    - yarn bazel test //tools/ts-api-guardian:all //tools/public_api_guard/... //packages/language-service/test //packages/compiler-cli/ngcc/test:test //packages/compiler-cli/integrationtest:integrationtest //packages/compiler-cli/test/compliance:compliance //packages/compiler/test //packages/compiler-cli/test:ngc //packages/compiler-cli/test/ngtsc:ngtsc
+    # At the moment 'browser:chromium-local' are to be broken in CI while locally they work
```
_are to be broken_ --> _are broken_ (?)

gkalpak reviewed (approved):
Commit message typos:
_as they C++ compilation is required_ --> _as C++ compilation is required_ (?)
_has no effect it is passed in both_ --> _has no effect if it is passed in both_

alan-agius4 committed:
ci: include more tests in windows ci

We added most of the tests excluded the once which are tagged with `browser:chromium-local` as they are currently not working in CI but work locally.

The `--test_tag_filters` option is now passed via command line as it seems that it has no effect as it is passed in both RC and command line. IE command line options are not merged with the RC options but rather overridden.

alan-agius4 commented:
Thanks @gkalpak for the review. PR cleaned-up

alan-agius4 committed:
ci: increase codefresh memory limit to decrease flakyness of known big tests

Such tests are:
```
//packages/language-service/test:test
//packages/bazel/test/ngc-wrapped:ngc_test
```

filipesilva commented on .codefresh/codefresh.yml:
```diff
@@ -12,13 +12,16 @@ steps:
     # Run tests
-    - yarn test-ivy-aot //packages/animations/test //packages/common/test //packages/forms/test //packages/http/test //packages/platform-browser/test //packages/platform-browser-dynamic/test //packages/router/test
-    - yarn bazel test //tools/ts-api-guardian:all //tools/public_api_guard/... //packages/language-service/test //packages/compiler-cli/ngcc/test:test //packages/compiler-cli/integrationtest:integrationtest //packages/compiler-cli/test/compliance:compliance //packages/compiler/test //packages/compiler-cli/test:ngc //packages/compiler-cli/test/ngtsc:ngtsc
+    # At the moment 'browser:chromium-local' are broken in CI while locally they work
```
Do you know why this happens?

alan-agius4 replied:
I haven’t look at this yet. I though of trying to figure that out for later.

filipesilva reviewed (approved):
Pretty sweet tbh, great work!

filipesilva commented:
@alxhub PTAL at the small change that needs compiler approval.
